esmvalcore/_recipe.py

```diff
@@ -1161,6 +1227,9 @@ def _initialize_variables(self, raw_variable, raw_datasets):
                 activity = get_activity(variable)
                 if activity:
                     variable['activity'] = activity
+            if 'timerange' in variable:
```

Comment:
Would it be possible to convert `start_year` and `end_year` to a timerange here and then remove `start_year` and `end_year` from the variable? That way there would be one consistent place for time information, instead of having to deal with two possible forms of the same information throughout the code.

Reply:
What I meant was: would it be possible to remove `start_year` and `end_year` everywhere from the code, i.e. do here something like this for backwards compatibility:
```python
if 'start_year' in variable and 'end_year' in variable:
    variable['timerange'] = f'{start_year}/{end_year}'
elif 'start_year' in variable:
    variable['timerange'] = f'{start_year}/{start_year}'
elif 'end_year' in variable:
    variable['timerange'] = f'{end_year}/{end_year}'
variable.pop('start_year', None)
variable.pop('end_year', None)
```
(not sure if I got the timerange syntax right) and then clean up the rest of the code so it no longer uses `variable['start_year']` and `variable['end_year']` anywhere anymore?

Reply:
Wouldn't that break `extract_time`?

Reply:
No, if users want to use `extract_time` they can configure it like any other preprocessor function from the recipe: https://esmvaltool--1133.org.readthedocs.build/projects/ESMValCore/en/1133/recipe/preprocessor.html#extract-time

Reply:
I don't see how storing the three tags is that bad though. Having to split the timerange value across the code everytime you need to work with just the years is more work than having all three tags in `variable`.

Reply:
Because it makes the time handling code needlessly complicated. I don't see why we would want to work with the years at all if we have a timerange. As far as I'm aware, `start_year` and `end_year` were used to do the following things:
- checking that the input files cover the requested timerange (in _recipe_checks.py) -> this can be done better with timerange
- filtering the list of preprocessor input files (in _data_finder.py and esgf/_search.py) -> this can be done better with timerange
- if the files contain more data than requested, clipping the data to the requested period using the `clip_start_end_year` preprocessor function -> you already implemented this with `clip_timerange`
- creating the preprocessor output filename -> you already implemented that this can be done with timerange

Or did I miss something?

Reply:
Ok I will try to update it accordingly, but when timeranges are given as periods ('*/P2Y') it's useful to have a start and end year.

Reply:
Thanks, that would be great! You already expand those wildcard style timeranges for the filled recipe, right? You can probably use the same code to convert that to a start and end datetime. The big advantage (in addition to simpler code) of using timerange for e.g. selecting the right input files is that we can just as easily support times that are stored with smaller time intervals per file than years.

Reply:
> Thanks, that would be great! You already expand those wildcard style timeranges for the filled recipe, right? You can probably use the same code to convert that to a start and end datetime. The big advantage (in addition to simpler code) of using timerange for e.g. selecting the right input files is that we can just as easily support times that are stored with smaller time intervals per file than years.

That is quickly doable when your timerange value looks something like this `19600101/19621231'. But I am having some issues when the timerange includes periods such as `19600101/P3Y2M1D`.  Adapting the file filtering to work with periods is not that small of a change and I'm not sure I can have it ready for the release.

Reply:
I am a bit confused about the esgf part. Does this part of the code happen before or after the download :

```
if 'start_year' in variable and 'end_year' in variable:
    variable['timerange'] = f'{start_year}/{end_year}'
elif 'start_year' in variable:
    variable['timerange'] = f'{start_year}/{start_year}'
elif 'end_year' in variable:
    variable['timerange'] = f'{end_year}/{end_year}'
variable.pop('start_year', None)
variable.pop('end_year', None)

```